build-tools/packages/build-cli/docs/typetestDetails.md

```diff
@@ -0,0 +1,151 @@
+# Further information about `generate typetests`
+
+The `generate typetests` command generates type compatibility tests based on the individual package settings in
+package.json.
+
+## What are type compatibility tests?
+
+Type compatibility tests are automatically generated test cases that check for type incompatibilities across package
+versions. They work by comparing types in the current version to the types in a previous version. The generated tests
+use the previous type in place of the new type, and vice-versa.
+
+See the [API Type Validation][] page in the Fluid Framework wiki for more information about type compatibility tests,
+including how to annotate expected type incompatibilities to fix failing tests.
+
+## Configuring type tests
+
+Type test generation is primarily configured in the package.json file of the package being tested, in the
+`typeValidation` section:
+
+```json
+"typeValidation": {
+  "disabled": false,
+  "version": "1.1.0",
+  "broken": {
+    "InterfaceDeclaration_AzureClientProps": {
+      "forwardCompat": false
+    }
+  }
+}
+```
+
+The `broken` section is used to indicate known breaking changes. Type tests can be completely disabled for a package
+using the `disabled` property. See [API Type Validation][] for more information.
+
+Generating type tests has two parts: _preparing package.json_ and _generating test modules_. By default, both steps are
+run for each package. You can run only one part at a time using the `--prepare` and `--generate` flags.
+
+[api type validation]: https://github.com/microsoft/FluidFramework/wiki/API-Type-Validation
```

Comment:
Curious on your thoughts WRT conventions for where to put link aliases. I find it useful to be able to look in a single place in a given document for the links so I know what things I can re-use and what I need to define as I go.

Would it make sense to move this to the bottom under a "links" comment like we generate for content on the website?

Feel free to disregard in the context of the PR - it just took me a few seconds to find where this was defined as I was reviewing, since I wanted to make sure it was defined somewhere.

Reply:
I usually put them at the end of the section in which they're used unless they're used in several places. Reason being that sections get moved/reorganized, and links sometimes break when they're far away from the content. But I don't care too much.